api/metric/api.go

```diff
@@ -62,20 +81,25 @@ func WithUnit(unit unit.Unit) Option {
 	}
 }
 
-// WithKeys applies the provided dimension keys.
+// WithKeys applies recommended dimension keys.  Multiple `WithKeys`
+// options accumulate.  The keys specified in this way are taken as
+// the recommended aggregation keys for Gauge, Cumulative, and
+// Additive metrics.  For Measure metrics, the keys recommended here
+// are taken as the default for aggregations.
 func WithKeys(keys ...core.Key) Option {
 	return func(m *Handle, _ *[]registry.Option) {
-		m.Keys = keys
+		m.Keys = append(m.Keys, keys...)
 	}
 }
 
-func (mtype MetricType) String() string {
-	switch mtype {
-	case Gauge:
-		return "gauge"
-	case Cumulative:
-		return "cumulative"
-	default:
-		return "unknown"
+// WithAggregation applies a user-recommended aggregation to this
+// metric, useful particularly for Measure type metrics.
+func WithAggregation(aggr aggregation.Operator) Option {
```

Comment:
This is not compatible with OpenCensus stats which is a primary requirement for this, because we need to not allow the instrumentation plugin to define the aggregation.

Reply:
This is not a *definition*. This is a recommendation. The SDK is not bound to follow this advice. If this value is not set, the default aggregation for MeasureMetrics is `None`. I like this because the application programmer can suggest `None` as a good aggregation for measure metrics, but could also suggest `Distribution`.